docs/patterns.rst

```diff
@@ -114,15 +118,119 @@ a ``categorical`` slot.
+
+A ``TextSlot`` can have any value, but it only has a single feature. It can be set, in which case the feature 
+has a value ``(1)``, or if it is not set it will have a value ``(0)``. 
+
+A ``CategoricalSlot`` has a number of values each of which is a feature. Taking the example below, when the 
+``restaurant_availability`` slot is set Rasa Core will be able to determine whether or not the restaurant in question is 
+available and choose radically different actions to perform based on the value.
+
+.. code-block:: yaml
+
+    restaurant_availability:
+        type: categorical
+        values:
+        - unknown
+        - booked-out
+        - waiting-list
+        - available
+
+A ``Slot`` will be set by Rasa Core if its name and the name of the entity detected by the NLU module match. However, when we wish to alter the flow of a conversation we would explicitly set the value of the slot from a custom ``Action`` and 
+then use the next ``Turn`` of the conversation to check what feature of the slot is set.
+
+.. code-block:: python
+
+    class ActionMakeBooking(Action):
+
+        def run(self, dispatcher, tracker, domain):
+            restaurant_name=tracker.get_slot("restaurant_name")
+            location=tracker.get_slot("location")
+            num_people=tracker.get_slot("people")
+            date=tracker.get_slot("date")
+            # this will fetch the availability of the restaurant from your DB or an API
+            availability=restaurantService.check_availability(restaurant_name, location, num_people, date)
+            return [SlotSet("restaurant_availability", availability)]
+
+The snippet of code above from a hypothetical ``Action`` shows that the value of the slot ``restaurant_availability`` is determined by querying a database or API. The restaurant availability is not something that is known when we train the dialogue model, the ``Slot`` value is the only way we can alter the course of the conversation based on information from the outside world.
+
+The data fetched from an API call can also be stored for later use without altering the outcome of a conversation as detailed in :ref:`unfeaturized_slots`.
+
+**Slot Features Example**
+
+.. note:: 
+    These example stories have been constructed manually for illustrative purposes. While this is a valid approach to training your model the preferred approach is to use :ref:`supervised learning <tutorial_supervised>` which generates stories that are *much* less error-prone.
+
+In this first story we will try and make a booking for 5 people in a restaurant on the night of 21st August 2018. 
+In this case the restaurant is booked out so we want to apologize to the customer and suggests similar restaurants. It is assumed that the Rasa Core model has been trained to recognise a message like *"Book Murphys Bistro on August 21 for 5 people"*
+
+.. code-block:: md
+
+    # restaurant unavailable
+    * _make_booking[people=5, date="2018-08-21T19:30:00+00:00", restaurant_id=145]
```

Comment:
we have an updated format format for stories, https://core.rasa.ai/stories.html
so e.g. 
`* make_booking{"people":"5", "date":"2018-08-21T19:30:00+00:00", "restaurant_id":"145"}`